pandas-stubs/core/frame.pyi

```diff
@@ -555,7 +555,7 @@ class DataFrame(NDFrame, OpsMixin):
     @overload
     def __getitem__(  # type: ignore[misc]
         self,
-        key: Series[_bool]
+        key: Series
```

Comment:
One DataFrame test does `df[Series[str]]`

Reply:
So shouldn't we just do `Series[_bool] | Series[str]` to not widen it too much?

Reply:
I think it can be any Series:

```py
>>> pd.DataFrame({1: [0]})[pd.Series([1])]
   1
0  0
>>> pd.DataFrame({1.0: [0]})[pd.Series([1.0])]
   1.0
0    0
>>> x = object()
>>> pd.DataFrame({x: [0]})[pd.Series([x])]
   <object object at 0x7fa66230ff30>
0                                  0
```

Reply:
Sure, but do we want to encourage that?  I'd rather limit the possibilities than allow anything, because it is implying that we are allowing anything to be a column name.

Reply:
> implying that we are allowing anything to be a column name.

I think that is already the case `pd.DataFrame(..., columns: Axes)` and `Axes: TypeAlias = AnyArrayLike | list | dict | range | tuple`. The only potential harm I see is that some `__getitem__` overloads might become ambiguous but that doesn't seem the be the case.

Reply:
OK - let's go with your change with an untyped `Series`